app/src/org/commcare/views/notifications/NotificationMessage.java

```diff
@@ -24,18 +25,26 @@ public int describeContents() {
 
     @Override
     public void writeToParcel(Parcel dest, int flags) {
-        dest.writeStringArray(new String[]{category, title, details, actions});
+        String buttonText = buttonInfo != null ? buttonInfo.getButtonText() : "";

```

Comment:
I think a more scalable design here is to make `NotificationActionButtonInfo` implements  `Parcelable` and localise the code to write and load `NotificationActionButtonInfo` inside itself and then use `dest.writeParcelable();` and `source.readParcelable()` respectively in this class to write the button info.